File: data/readable_format/fortran/file465.f90
program test_recoverC
    implicit none

    ! Define constants directly in the program
    integer, parameter :: ME2c_max = 100
    integer, parameter :: nsh_max = 10

    real, dimension(ME2c_max) :: hlist, dhlist
    real, dimension(nsh_max, nsh_max) :: hbox, dhbox
    integer :: n1, n2
    integer :: i, j

    ! Initialize arrays to 0
    hlist = 0.0
    dhlist = 0.0
    hbox = 0.0
    dhbox = 0.0

    ! Initialize test case
    n1 = 2
    n2 = 3
    ! Assign values to the first 6 elements of hlist and dhlist
    hlist(1:6) = [1.0, 2.0, 3.0, 4.0, 5.0, 6.0]
    dhlist(1:6) = [0.1, 0.2, 0.3, 0.4, 0.5, 0.6]

    ! Call the subroutine
    call recoverC(n1, n2, hlist, dhlist, hbox, dhbox)

    ! Output results for verification
    do i = 1, n1
        do j = 1, n2
            print *, "hbox(", i, ",", j, ") = ", hbox(i, j)
            print *, "dhbox(", i, ",", j, ") = ", dhbox(i, j)
        end do
    end do

contains

    subroutine recoverC (n1, n2, hlist, dhlist, hbox, dhbox)
        implicit none

        integer, intent (in) :: n1, n2

        real, intent(in), dimension (ME2c_max) :: hlist 
        real, intent(in), dimension (ME2c_max) :: dhlist 

        real, intent(out), dimension (nsh_max, nsh_max) :: hbox
        real, intent(out), dimension (nsh_max, nsh_max) :: dhbox

        integer :: index
        integer :: indexcoulomb
        integer :: ii
        integer :: kk

        ii = 1
        kk = 0

        indexcoulomb = n1*n2
        do index = 1, indexcoulomb
            kk = kk + 1
            hbox(ii,kk) = hlist(index)
            dhbox(ii,kk) = dhlist(index)
            if (mod(index,n2) .eq. 0) then
                ii = ii + 1
                kk = kk - n2
            end if
        end do

    end subroutine recoverC

end program test_recoverC

File: data/readable_format/cpp/file465.cpp
#include <iostream>
#include <vector>
#include <cassert>

// Assuming dimensions and interactions constants
const int ME2c_max = 100;
const int nsh_max = 10;

// Function declaration and definition in the same file
void recoverC(int n1, int n2, const std::vector<float>& hlist, const std::vector<float>& dhlist,
              std::vector<std::vector<float>>& hbox, std::vector<std::vector<float>>& dhbox) {
    int index;
    int indexcoulomb;
    int ii;
    int kk;
    
    ii = 0;  // Adjusted for 0-based indexing
    kk = -1; // Adjusted for 0-based indexing and incremented before use

    indexcoulomb = n1 * n2;
    for (index = 0; index < indexcoulomb; ++index) { // Adjust loop for 0-based indexing
        kk++;
        hbox[ii][kk] = hlist[index];
        dhbox[ii][kk] = dhlist[index];
        if ((index + 1) % n2 == 0) { // Adjust condition for 0-based indexing
            ii++;
            kk -= n2;
        }
    }
}

int main() {
    int n1 = 2;
    int n2 = 3;
    std::vector<float> hlist = {1.0, 2.0, 3.0, 4.0, 5.0, 6.0};
    std::vector<float> dhlist = {0.1, 0.2, 0.3, 0.4, 0.5, 0.6};
    std::vector<std::vector<float>> hbox(n1, std::vector<float>(n2));
    std::vector<std::vector<float>> dhbox(n1, std::vector<float>(n2));

    // Call the function
    recoverC(n1, n2, hlist, dhlist, hbox, dhbox);

    // Simple test to verify output
    for (int i = 0; i < n1; ++i) {
        for (int j = 0; j < n2; ++j) {
            std::cout << "hbox[" << i << "][" << j << "] = " << hbox[i][j] << ", ";
            std::cout << "dhbox[" << i << "][" << j << "] = " << dhbox[i][j] << std::endl;
            // Add your verification logic here, for example, using assert
        }
    }

    return 0;
}